tokio/src/runtime/handle.rs

```diff
@@ -201,6 +201,18 @@ impl Handle {
         let _ = self.blocking_spawner.spawn(task, &self);
         handle
     }
+
+    /// TODO: write docs if this is a good direction
+    pub fn block_on<F: Future>(&self, future: F) -> F::Output {
+        // Enter the **runtime** context. This configures spawning, the current I/O driver, ...
+        let _rt_enter = self.enter();
+
+        // Enter a **blocking** context. This prevents blocking from a runtime.
+        let mut _blocking_enter = crate::runtime::enter(true);
+
+        // Block on the future
+        _blocking_enter.block_on(future).expect("failed to park thread")
```

Comment:
Then this should be a block on that runs the task and can wake up on runtime shutdown. We likely wanna use the same notify trick?

Reply:
The future isn't a task per se. When the runtime is shutdown, in *theory* all resources should transition to a "shutdown" state and in-flight operations canceled. This would result in this block_on waking up.

Reply:
Even if the task is doing nothing? E.g. if I just call `handle.block_on(std::future::pending())`, then what?

Reply:
Hmm, I guess my gut is that calling `Handle::block_on` shouldn't necessarily force exit when the runtime shuts down. My gut reaction is that `handle.block_on(std::future::pending())` would just run indefinitely.